debezium-connector-mysql/src/main/java/io/debezium/connector/mysql/rest/DebeziumMySqlConnectorResource.java

```diff
@@ -49,41 +40,13 @@ public String getSchemaFilePath() {
     }
 
     @Override
-    public Connector getConnector() {
+    public BaseSourceConnector getConnector() {
         return new MySqlConnector();
     }
 
     @Override
     public List<DataCollection> getMatchingCollections(Configuration configuration) {
-        final MySqlConnectorConfig config = new MySqlConnectorConfig(configuration);
-        try (AbstractConnectorConnection connection = config.getConnectorAdapter().createConnection(configuration)) {
-            Set<TableId> tables;
-
-            final List<String> databaseNames = new ArrayList<>();
-            connection.query("SHOW DATABASES", rs -> {
-                while (rs.next()) {
-                    databaseNames.add(rs.getString(1));
-                }
-            });
-
-            List<DataCollection> allMatchingTables = new ArrayList<>();
-
-            for (String databaseName : databaseNames) {
-                if (!config.getTableFilters().databaseFilter().test(databaseName)) {
-                    continue;
-                }
-                tables = connection.readTableNames(databaseName, null, null, new String[]{ "TABLE" });
-                var matchingTables = tables.stream()
-                        .filter(tableId -> config.getTableFilters().dataCollectionFilter().isIncluded(tableId))
-                        .map(tableId -> new DataCollection(tableId.catalog(), tableId.table()))
-                        .collect(Collectors.toList());
-                allMatchingTables.addAll(matchingTables);
-            }
-            return allMatchingTables;
-        }
-        catch (SQLException e) {
-            throw new DebeziumException(e);
-        }
+        return getConnector().getMatchingCollections(configuration);
```

Comment:
I don't think so (if I got you right) because `getConnector()` and `getMatchingCollections()` have entirely connector specific code. There is not much in common or else we have to add a bunch of other methods to the interface to provide the diverging `*ConnectorConfig` instances `*Connection` classes etc.

Reply:
never mind, I think it looks very good now!